test/dev-tools/smoke.js

```diff
@@ -0,0 +1,12 @@
+'use strict';
+
+const assert = require('assert');
+
+const {SmokeRunner} = require('./../../dev-tools/smoke');
+
+describe('SmokeRunner', async () => {
+	it('should run successfully', async () => {
+		const opts = {format: 'gen7doublescustomgame', maxGames: 1, prng: [1, 2, 3, 4]};
+		assert.strictEqual(await (new SmokeRunner(opts).run()), 0);
```

Comment:
How much time does this take?

Reply:
Running a single battle? 100ms.

Reply:
On _your_ computer. Poor Kris, on the other hand...

Reply:
Kris is busy adding even slower tests with his team validation :) https://github.com/Zarel/Pokemon-Showdown/pull/5397#discussion_r271467476

WDYT about @Slayer95's idea of having a separate option for slow tests (which Travis or people with servers under their desk to run)? I'd want to run a whole cycle of the smoke tests (~1 minute) instead of just a single battle if we had that.

Reply:
Yeah, I like the idea of having a separate option for slow tests. But we're already stretching Travis CI to its limit, so it'd definitely just be something we'd have to use a different server for.

Also I hear Travis CI is imploding. We should probably look into switching. I hear Azure Pipelines is a thing? Also Circle CI? @Slayer95, can you investigate?

Reply:
> Also I hear Travis CI is imploding.

That seems to point to last week's outage, which is the only major issue in the last 4 months as per https://www.traviscistatus.com/

They are yet to publish a postmortem about it though.

> But we're already stretching Travis CI to its limit

I have noticed some delay in Travis yea. But is that some actual limit as opposed to the direct effect of our longer build time?

> Azure Pipelines is a thing? Also Circle CI? @Slayer95, can you investigate?

I guess I'll have a look.

Reply:
> I have noticed some delay in Travis yea. But is that some actual limit as opposed to the direct effect of our longer build time?

Kris pointed out he got a timeout in Travis when he added the extra BF validator.

Reply:
That should be the default test timeout from ``mocha`` itself...,

Reply:
Are we blocking on this? I can disable both `test/dev-tools/{smoke,harness}.js` tests if you're concerned (I don't know if you saw it in the previous PR, because while a single smoke test is slower, theyre both around the same O(100ms) run time?)

Reply:
We're not blocking on anything. I'm not opposed to merging and figuring out the rest later.